### ISSUE TITLE
Unable to sign evm transactions again (regression)

### ISSUE THREAD

**Issue by randy75828** [chainapsis/keplr-chain-registry#753]
It seems like there is a recent removal in the evm field in carbon.json as shown here:

https://github.com/chainapsis/keplr-chain-registry/commit/feeaefb33fd95d18c2108cf5392fc893695f4c46

and that the evm info for the chain is moved to:

https://github.com/chainapsis/keplr-chain-registry/blob/main/evm/eip155%3A9790.json

but this brings back this error `Not ethermint like and EVM chain` due to the missing evmInfo field. 

This sudden update is breaking our chains evm transaction. Can anyone help?

**Comment by randy75828** [comment]
@Thunnini  @delivan @claraexmachina

**Comment by delivan** [comment]
Hey @randy75828 It would be helpful to tell what and where do you try to sign EVM transaction.

**Comment by randy75828** [comment]
I'm signing on our demex application. It is a feature that is planned for release today but due to the sudden change in the chain config here it is blocked in testing environment.

In short it is blocked because of the evm info removal 

@delivan

**Comment by randy75828** [comment]
I saw that reversion is probably due to the SWTH values not showing corrrectly on the keplr wallet, which probably stems from the fact that probably the wallet cannot differentiate an 8dp SWTH (cosmos) vs 18dp SWTH (evm). However this reversion is causing us to not being able to sign evm transaction

**Comment by delivan** [comment]
You are right. and another crucial problem is Metamask user's address is derived with 'coin type 60', but Keplr user's address is derived with 'coin type 118'. We don't support both coin typ 60 and 118 as on one chain. so we split Carbon chain config into cosmos version(coin type 118, cosmos/carbon.json) and evm version (coin type 60, evm/eip155:9790.json).

If you want to use evm functionality, you should use Keplr via `window.keplr.ethereum` like using Metamask in your app.

**Comment by randy75828** [comment]
<img width="344" alt="image" src="https://github.com/user-attachments/assets/ca66f19c-770f-4852-9a20-3069f32d0d7f">
seems like carbon chain is not in the list? how do i add it in?

**Comment by delivan** [comment]
If you want to sign and send a transaction, you can use `eth_sendTransaction`. it includes signing process. 
`eth_signTransaction` is not supported for now but it will be supported asap. @randy75828

**Comment by randy75828** [comment]
@delivan i realised there is no way for me to suggest chain unless i hardcode the json i got while added keplr via the official keplr website

`https://chains.keplr.app/`
```
{
  "rpc": "https://evm-api.carbon.network",
  "nodeProvider": {
    "name": "Switcheo Labs",
    "email": "info@switcheo.network",
    "website": "https://www.switcheo.com/"
  },
  "chainId": "eip155:9790",
  "chainName": "Carbon EVM",
  "chainSymbolImageUrl": "https://raw.githubusercontent.com/chainapsis/keplr-chain-registry/main/images/eip155:9790/chain.png",
  "bip44": {
    "coinType": 60
  },
  "currencies": [
    {
      "coinDenom": "SWTH",
      "coinMinimalDenom": "swth",
      "coinDecimals": 18,
      "coinGeckoId": "switcheo"
    }
  ],
  "feeCurrencies": [
    {
      "coinDenom": "SWTH",
      "coinMinimalDenom": "swth",
      "coinDecimals": 18,
      "coinGeckoId": "switcheo"
    }
  ],
  "rest": "https://evm-api.carbon.network",
  "evm": {
    "chainId": 9790,
    "rpc": "https://evm-api.carbon.network"
  },
  "features": [
    "eth-address-gen",
    "eth-key-sign"
  ],
  "beta": true
}
```

is  there any api available for me to call?

**Comment by delivan** [comment]
You can call `wallet_addEthereumChain` with some params via `window.keplr.ethereum.request` method.
I refer a [document from Metamask](https://docs.metamask.io/wallet/reference/wallet_addethereumchain/) which is working same with Keplr @randy75828

**Comment by randy75828** [comment]
Hmm but in that case I'll need to hardcore the params regardless, based on what is on  keplr wallet registry for my Evm chain?

**Comment by delivan** [comment]
Yes, It would be good to do that and one thing you should change is `chainId` value as "0xfef" not "eip155:9790"

**Comment by randy75828** [comment]
thanks
<img width="414" alt="image" src="https://github.com/user-attachments/assets/83a198a2-bc68-48d9-8801-3c8ed41438a0">

for this, is it possible to suggest Carbon EVM straightaway?

**Comment by delivan** [comment]
There is no way for that connection popup, but after that you can call `wallet_switchEthereumChain` to switch to your chain to connect.
Refer the [document](https://docs.metamask.io/wallet/reference/wallet_switchethereumchain/) for it

**Comment by randy75828** [comment]
Thanks

<img width="360" alt="image" src="https://github.com/user-attachments/assets/7daeba98-caa2-49fe-91d4-6d2ce76b345e">
i am getting this error even though the chain has been added and switched, do u know why

**Comment by delivan** [comment]
@randy75828 Have you connected to your chain? does that error make your logic stuck?

**Comment by randy75828** [comment]
I think that happened when i remove the chain from the extension. Can only resolve it when i disconnect the wallet from my website via the keplr extension

**Comment by randy75828** [comment]
i figured the exception was thrown because Carbon EVM chain coin type is 60
<img width="195" alt="image" src="https://github.com/user-attachments/assets/0a717c63-e589-4a85-b376-36768bdca75c">
is it possible for me to change to 118

i am getting another validation error when it is 118. is there a way to allow 118?

<img width="604" alt="image" src="https://github.com/user-attachments/assets/b9fa8ef0-9ed5-4ae1-a402-a4a50145bf01">

@delivan

**Comment by randy75828** [comment]
Found a way to patch it


https://github.com/chainapsis/keplr-chain-registry/pull/756
 
 @delivan can u probably take a look at this PR?
